homeassistant/helpers/filter.py

```diff
@@ -0,0 +1,125 @@
+                self.filter = self._outlier
+        else:
+            self.logger.error("Unknown filter <%s>", filter_algorithm)
+            return
+
+    def __call__(self, func):
+        """Decorate function as filter."""
+        states = self.states
+        filter_algo = self.filter
+        filter_args = self.filter_args
+
+        if len(self.states):
+            last_state = self.states[-1]
+        else:
+            last_state = None
+
+        def func_wrapper(self):
+            """Wrap for the original state() function."""
+            new_state = func(self)
+            try:
+                Filter.logger.debug("Filter arguments: %s", filter_args)
+                filtered_state = filter_algo(new_state=float(new_state),
+                                             states=states, **filter_args)
+            except TypeError:
+                return None
+            except ValueError as e:
+                Filter.logger.warning("Invalid Value in %s: %s, reason: %s",
+                                      self.entity_id, float(new_state), e)
+                return last_state
+            states.append(filtered_state)
+            Filter.logger.debug("%s(%s) -> %s", filter_algo.__name__,
+                                new_state, filtered_state)
+            return filtered_state
+
+        return func_wrapper
+
+    @staticmethod
+    def _outlier(new_state, states, **kwargs):
+        """BASIC outlier filter.
+
+        Will through a ValueError indicating an outlier value
+        that should not be published, else the current value.
+
+        Args:
+            new_state (float): new value to the series
+            states (deque): previous data series
+            constant (int): stdev multiplier.
+
+        Returns:
+            the original new_state
+
+        """
+        constant = kwargs.pop('constant', 0.10)
+
+        if (len(states) > 1 and
+                abs(new_state - statistics.median(states)) >
+                constant*statistics.median(states)):
+            raise ValueError("Outlier detected")
+        return new_state
+
+    @staticmethod
+    def _lowpass(new_state, states, **kwargs):
+        """BASIC Low Pass Filter.
+
+        Args:
+            new_state (float): new value to the series
+            states (deque): previous data series
+            time_constant (int): time constant.
+
+        Returns:
+            a new state value that has been smoothed by filter
+
+        """
+        time_constant = kwargs.pop('time_constant', 4)
+        if len(kwargs) != 0:
+            Filter.logger.error("unrecognized params passed in: %s", kwargs)
+
+        try:
+            B = 1.0 / time_constant
+            A = 1.0 - B
+            filtered = A * states[-1] + B * new_state
+        except IndexError:
+            # if we don't have enough states to run the filter
+            # just accept the new value
+            filtered = new_state
+        return round(filtered, 2)
```

Comment:
I don't really like this (`2`). With some sensors, sure I want two digit accuracy, for example a DHT22. But with others, like a barometer, this would just add unnecessary precision + it makes the frontend ugl(ier). I get that too many config options can be a bit overwhelming to users, but this just seems like an important option to me.

Reply:
forgot that "2"... should be a configurable option

I would like nonetheless to have "sensible defaults"

Reply:
Also, I like having my sensors have a consistent number of digits. For example, if my sensor outputs 20.00 °C I want it to be displayed with two zeros and not one. I am not sure whether this platform does this already, but posting it here as a potential suggestion.

Reply:
> I would like nonetheless to have "sensible defaults"

Sure!